kamon-scala-future/src/main/scala/kamon/scala/FutureSpans.scala

```diff
@@ -0,0 +1,42 @@
+package kamon.scala
+
+import kamon.Kamon
+import kamon.Kamon.withContextKey
+import kamon.trace.Span
+import kamon.util.CallingThreadExecutionContext
+
+import scala.concurrent.Future
+
+object FutureSpans {
+  def withNewAsyncSpan[T](operationName: String, spanKind: String, component: String, tags: Map[String, String] = Map.empty)(code: Span ⇒ Future[T]): Future[T] = {
+    val spanBuilder = Kamon.buildSpan(operationName)
```

Comment:
Be careful that this span's parent will be set to w/e is in the current scope, not in the scope of which the Future body is run.
For example, test it with akka-http's route and you should that the parent span isn't being set correctly because the parent span isn't available inside directives.

```
get {
  // In this scope there's no parent span
  onComplete(withNewAsyncSpan(...)) {
  }
}
``` 

This is why I had to wrap everything inside a `Future.apply`
https://github.com/kamon-io/kamon-futures/issues/3#issuecomment-468941788

Reply:
I'm not sure if this is related or not, but we noticed that there were some issues with Context propagation and the Future directives related to the fact that we were not instrumenting `FastFuture`.. that was fixed already on the Akka HTTP instrumentation and maybe it would yield a different (and expected) behavior in this case.

Reply:
That could be the cause. Although for types like cats-effect `IO` I think the wrapping is required.